src/app/view/editor-main-view/data-views/trainrunsections.view.ts

```diff
@@ -50,19 +63,138 @@ export class TrainrunSectionsView {
     const t1: Vec2D = pathVec2D[2];
     const diff: Vec2D = Vec2D.sub(t1, s1);
     let a: number = (Math.atan2(diff.getY(), diff.getX()) / Math.PI) * 180.0;
-    if (Math.abs(a) > 90) {
+    if (Math.abs(a) > ANGLE_UPSIDE_DOWN_THRESHOLD) {
       a = a + 180;
     }
     // Math.atan2 -> edge cases -> correct manually
-    if (Math.abs(diff.getX()) < 1) {
-      a = -90;
+    if (Math.abs(diff.getX()) < EDGE_CASE_THRESHOLD) {
+      a = DEFAULT_ANGLE_VERTICAL;
     }
-    if (Math.abs(diff.getY()) < 1) {
-      a = 0;
+    if (Math.abs(diff.getY()) < EDGE_CASE_THRESHOLD) {
+      a = DEFAULT_ANGLE_HORIZONTAL;
     }
     return "translate(" + x + "," + y + ") rotate(" + a + ", 0,0) ";
   }
 
+  /**
+   * Version that works with TrainrunSectionViewObject to handle custom paths for collapsed chains
+   * This method needs access to the view instance to get the collapsed path
+   */
+  static translateAndRotateTextForViewObject(
```

Comment:
Ditto for this function, as well as `getTextPositionsForCollapsedChain`, `getAdditionPositioningValueForViewObjectWithCollapsedSupport` and `translateAndRotateTextForViewObjectWithCollapsedSupport`. We shouldn't need to duplicate all of the logic here.